src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/WebServicesTestController.cs

```diff
@@ -0,0 +1,216 @@
+                mockName);
+        }
+
+        public void RunPsTestWorkflow(
+            Func<string[]> scriptBuilder,
+            Action<LegacyTest.CSMTestEnvironmentFactory> initialize,
+            Action cleanup,
+            string callingClassType,
+            string mockName)
+        {
+            var providers = new Dictionary<string, string>
+            {
+                { "Microsoft.Resources", null },
+                { "Microsoft.Features", null },
+                { "Microsoft.Authorization", null }
+            };
+            var providersToIgnore = new Dictionary<string, string>
+            {
+                {"Microsoft.Azure.Management.Resources.ResourceManagementClient", "2016-02-01"}
+            };
+            HttpMockServer.Matcher = new PermissiveRecordMatcherWithApiExclusion(true, providers, providersToIgnore);
+
+            HttpMockServer.RecordsDirectory = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "SessionRecords");
+            using (var context = MockContext.Start(callingClassType, mockName))
+            {
+                this.csmTestFactory = new LegacyTest.CSMTestEnvironmentFactory();
+                if (initialize != null)
+                {
+                    initialize(this.csmTestFactory);
+                }
+
+                this.SetupManagementClients(context);
+                helper.SetupEnvironment(AzureModule.AzureResourceManager);
+
+                var callingClassName = callingClassType
+                                      .Split(new[] { "." }, StringSplitOptions.RemoveEmptyEntries)
+                                      .Last();
+
+                helper.SetupModules(AzureModule.AzureResourceManager,
+                   "ScenarioTests\\Common.ps1",
+                   "ScenarioTests\\" + callingClassName + ".ps1",
+                   helper.RMProfileModule,
+                   helper.RMResourceModule,
+                   helper.GetRMModulePath(@"AzureRM.MachineLearning.psd1"),
+                   "AzureRM.Storage.ps1");
+
+                try
+                {
+                    if (scriptBuilder != null)
+                    {
+                        var psScripts = scriptBuilder();
+
+                        if (psScripts != null)
+                        {
+                            helper.RunPowerShellTest(psScripts);
+                        }
+                    }
+                }
+                finally
+                {
+                    if (cleanup != null)
+                    {
+                        cleanup();
+                    }
+                }
+            }
+        }
+
+        private void SetupManagementClients(MockContext context)
+        {
+            this.ResourceManagementClient = LegacyTest.TestBase.GetServiceClient<ResourceManagementClient>(this.csmTestFactory);
+            this.WebServicesManagementClient = context.GetServiceClient<AzureMLWebServicesManagementClient>(TestEnvironmentFactory.GetTestEnvironment());
```

Comment:
@markcowl I will try to make the line break at between 80 - 100. But is there any tools or extensions to do it automatically?